components/ai_chat/ai_chat_api_unittest.cc

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/ai_chat_api.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/functional/callback.h"
+#include "base/test/task_environment.h"
+#include "base/test/values_test_util.h"
+#include "services/data_decoder/public/cpp/test_support/in_process_data_decoder.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+
+namespace ai_chat {
+
+class AIChatAPIUnitTest : public testing::Test {
+ public:
+  AIChatAPIUnitTest()
+      : shared_url_loader_factory_(
+            base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+                &url_loader_factory_)) {
+    ai_chat_api_ = std::make_unique<AIChatAPI>(shared_url_loader_factory_);
+  }
+
+  ~AIChatAPIUnitTest() override = default;
+
+  void SendMesage(AIChatAPI::ResponseCallback callback,
+                  const std::string& text) {
+    ai_chat_api_->set_response_callback_for_testing(callback);
+    ai_chat_api_->SendDataForTesting(text);
+  }
+
+ protected:
+  std::unique_ptr<AIChatAPI> ai_chat_api_ = nullptr;
+
+ private:
+  base::test::TaskEnvironment task_environment_;
+  network::TestURLLoaderFactory url_loader_factory_;
+  scoped_refptr<network::SharedURLLoaderFactory> shared_url_loader_factory_;
+  data_decoder::test::InProcessDataDecoder in_process_data_decoder_;
+};
+
+TEST_F(AIChatAPIUnitTest, ParseJson) {
+  base::RunLoop run_loop;
+  SendMesage(base::BindRepeating(
+                 [](base::RunLoop* run_loop, const std::string& response) {
+                   EXPECT_EQ(" Hello there!", response);
+                   run_loop->Quit();
+                 },
+                 &run_loop),
+             "data: {\"completion\": \" Hello there!\", \"stop\": null}");
+  run_loop.Run();
+
+  base::RunLoop run_loop2;
+  SendMesage(
+      base::BindRepeating(
+          [](base::RunLoop* run_loop, const std::string& response) {
+            EXPECT_EQ(" Hello there! How are you?", response);
+            run_loop->Quit();
+          },
+          &run_loop2),
+      "data: {\"completion\": \" Hello there! How are you?\", \"stop\": null}");
+  run_loop2.Run();
+
```

Comment:
Add a case for an unknown response schema, or missing JSON entirely

Reply:
Added